api/internal/managers/app/config/template.go

```diff
@@ -0,0 +1,25 @@
+package config
+
+import (
+	"bytes"
+	"fmt"
+)
+
+func (m *appConfigManager) processTemplate(configYAML string) (string, error) {
+	if configYAML == "" {
+		return configYAML, nil
+	}
+
+	tmpl, err := m.templateEngine.Parse(configYAML)
+	if err != nil {
+		return "", fmt.Errorf("template parse error: %w", err)
+	}
```

Comment:
why marshal and parse this each time we get the config? why not do it when we instantiate appConfigManager and store it as a property?